201-redis-vnet-geo-replication/azuredeploy.json

```diff
@@ -0,0 +1,147 @@
+            }
+        },
+        "existingPrimaryCacheSubnetName": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the existing primary cache subnet."
+            }
+        },
+        "newPrimaryCacheName": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the new primary cache."
+            }
+        },
+        "secondaryLocation": {
+            "type": "string",
+            "metadata": {
+                "description": "The location of the existing secondary virtual network and new secondary cache."
+            }
+        },
+        "existingSecondaryVirtualNetworkName": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the existing secondary virtual network."
+            }
+        },
+        "existingSecondaryCacheSubnetName": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the existing secondary cache subnet."
+            }
+        },
+        "newSecondaryCacheName": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the new secondary cache."
+            }
+        }
+    },
+    "variables": {
+        "primarySubnetId": "[resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('existingPrimaryVirtualNetworkName'), parameters('existingPrimaryCacheSubnetName'))]",
+        "secondarySubnetId": "[resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('existingSecondaryVirtualNetworkName'), parameters('existingSecondaryCacheSubnetName'))]"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Cache/Redis",
+            "apiVersion": "2018-03-01",
+            "name": "[parameters('newPrimaryCacheName')]",
+            "location": "[parameters('location')]",
+            "properties": {
+                "sku": {
+                    "name": "Premium",
+                    "family": "P",
+                    "capacity": 1
+                },
+                "subnetId": "[variables('primarySubnetId')]"
+            },
+            "comments": "Create a Premium cache in the primary location in the primary virtual network."
+        },
+        {
+            "type": "Microsoft.Cache/Redis",
+            "apiVersion": "2018-03-01",
+            "name": "[parameters('newSecondaryCacheName')]",
+            "location": "[parameters('secondaryLocation')]",
+            "comments": "Create a Premium cache in the secondary location in the secondary virtual network.",
+            "properties": {
+                "sku": {
+                    "name": "Premium",
+                    "family": "P",
+                    "capacity": 1
+                },
+                "subnetId": "[variables('secondarySubnetId')]"
+            }
+        },
+        {
+            "type": "Microsoft.Resources/deployments",
```

Comment:
why is this nested?

Reply:
Without nesting it, validation fails on the template with the error:

```json
{
  "error": {
    "code": "InvalidTemplate",
    "details": null,
    "message": "Deployment template validation failed: 'The template resource 'Microsoft.Cache/Redis/contoso-primary-cache/linkedServers/contoso-secondary-cache' cannot reference itself. Please see https://aka.ms/arm-template-expressions/#reference for usage details.'.",
    "target": null
  },
  "properties": null
}

Reply:
I don't see a self-reference anywhere, let's remove the nesting and debug it...

Reply:
I removed the nesting in commit a50808f.